lib/src/network_manager_client.dart

```diff
@@ -2496,6 +2496,39 @@ class NetworkManagerClient {
         object ?? _NetworkManagerObject(_bus, DBusObjectPath('/'), {}));
   }
 
+  /// Activates a connection for [device].
+  ///
+  /// A specific [connection] may be specified, or else it is detected automatically.
+  ///
+  /// Note that when activating a wireless connection, the [accessPoint] must be specified.
+  Future<NetworkManagerActiveConnection?> activateConnection(
```

Comment:
I think this should be `Future<NetworkManagerActiveConnection>` and throw exceptions in the cases where this fails.